text/src/autogluon/text/automm/optimization/utils.py

```diff
@@ -57,7 +59,13 @@ def get_loss_func(
         else:
             loss_func = nn.CrossEntropyLoss()
     elif problem_type == REGRESSION:
-        loss_func = nn.MSELoss()
+        if loss_func_replaced is not None:
+            if "bcewithlogitsloss" in loss_func_replaced.lower():
+                loss_func = nn.BCEWithLogitsLoss()
+            else:
+                loss_func = nn.MSELoss()
+        else:
+            loss_func = nn.MSELoss()
```

Comment:
This implementation restricted using bce to only regression tasks. We can improve this code. Keep the original code and Insert the:
```
if loss_func_replaced is not None and "bcewithlogitsloss" in loss_func_replaced.lower():
    loss_func = nn.BCEWithLogitsLoss()
```
at the end of this function. In this way, users can also use bce loss for classification tasks.

Reply:
Yeah, how about to revise it in a later PR. I think BCE for classification has not been tested. We can fix the logic later.

Reply:
The current implementations in `transform_prediction` and `_compute_metric_score` seem not to support BCE for classification. Considering this limitation, we should raise error if detecting BCE in classification tasks.